mautic/mautic
# Option to set a strict transport security header in response

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [No]
| New feature/enhancement? (use the a.x branch)      | [Yes]
| Deprecations?                          | [No]
| BC breaks? (use the c.x branch)        | [No]
| Automated tests included?              | [Yes]
| Related user documentation PR URL      | #
| Related developer documentation PR URL | #
| Issue(s) addressed                     | #

#### Description:

This PR introduces the ability to set the Strict Transport Security header for responses in the system configuration - 
[Strict-Transport-Security docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security)

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to Configuration => System Settings (.../s/config/edit) and find the "Additional Response Headers" section.
![hsts_1](https://user-images.githubusercontent.com/102536220/196393911-2a583cde-3571-480d-8890-0c04344d2503.jpg)
4. Switch Strict Transport Security to Yes and set Include subdomains and Expire Time (seconds) according to your preferences.
5. Save the configuration. 
6. Check that the responses contain the appropriate Strict Transport Security header.
![hsts_2](https://user-images.githubusercontent.com/102536220/196397260-da040447-c44e-4d30-980a-c8bfb739c940.jpg)
